dev-docs/source/WritingAnAlgorithm.rst

```diff
@@ -0,0 +1,245 @@
+.. _WritingAnAlgorithm:
+
+Writing An Algorithm
+====================
+
+.. contents::
+  :local:
+
+Introduction
+############
+
+Mantid's `plugin <https://www.mantidproject.org/Plugin>`__ architecture has been engineered so that it is easy for a user 
+to write their own algorithm. This page is a primer for the user about to write their first algorithm and assumes no 
+great knowledge of C++. 
+It covers the basics, with links to more advanced options where appropriate. Note if you are looking to add a 
+`plugin <https://www.mantidproject.org/Plugin>`__ fit function rather than an algorithm then see 
+`Writing a Fit Function <https://www.mantidproject.org/Writing_a_Fit_Function>`__. 
+There is special description for the case when you are looking to add a custom `MD conversion plugin <WritingCustomConvertToMDTransformation>`__.
+
+Alternatively, you can implement your algorithm in `Python <https://www.mantidproject.org/Extending_Mantid_With_Python>`__. 
+See `Python Vs C++ Algorithms <https://www.mantidproject.org/Python_Vs_C%2B%2B_Algorithms>`__ for a comparison of Mantid's 
+two programming languages.
+
+All `algorithms <https://www.mantidproject.org/Algorithm>`__ in Mantid `inherit <http://en.wikipedia.org/wiki/Inheritance_(computer_science)>`__ 
+from a base Algorithm class, which provides the support and services required for running a specific 
+algorithm and greatly simplifies the process of writing a new one.
+
+Getting Started
+###############
+The first step is to create a new directory, with any name of your choice, under your MantidInstall directory
+(on Windows, probably located at C:\MantidInstall). Alternatively, you can just do everything in the 
```

Comment:
The `\` needs escaping, it doesn't render as it is. Paths should also be `raw text formatted`.